src/SFML/Graphics/Shape.cpp

```diff
@@ -275,14 +255,21 @@ void Shape::updateOutline()
         Vector2f p2 = m_vertices[index + 1].position;
 
         // Compute their normal
-        Vector2f n1 = computeNormal(p0, p1);
-        Vector2f n2 = computeNormal(p1, p2);
+        const auto findNormal = [](const sf::Vector2f& vector)
+        {
+            if (vector.lengthSq() == 0.f)
+                return vector;
```

Comment:
This case is unfortunately possible when `i` equals 0.
```cpp
        Vector2f p0 = (i == 0) ? m_vertices[count].position : m_vertices[index - 1].position;
        Vector2f p1 = m_vertices[index].position;
``` 
In that case `p0` and `p1` are both `m_vertices[index].position`. I feel like the loop index should start at 1 instead of 0 to prevent that. Unfortunately we haven't yet gotten around to testing `sf::Shape` so we can't lean on our unit tests to confirm that we didn't break anything.

Reply:
Yes, the `(i == 0)` condition is confusing, a more standard approach here would be
`m_vertices[(index - 1) % count].position`.

I guess it _is_ possible to have duplicate vertices, e.g. by using [`sf::ConvexShape`](https://www.sfml-dev.org/documentation/2.5.1/classsf_1_1ConvexShape.php) and manually inserting the same point twice. Which could already be seen as another invalidated variant, but we don't check this.

What does everyone think -- is this something worth hardening? If yes, we could open an issue so we don't forget it. If no, no action is needed.

Reply:
I don't think it's worth addressing, the checks would introduce some overhead and I don't think anyone has ever complained about this issue.